routes/export.rb

```diff
@@ -0,0 +1,32 @@
+require 'models/user'
+require 'models/resource'
+require 'models/permission'
+require 'models/event'
+require 'csv'
+require 'date'
+
+get '/export/?' do
+    user_events = Event.includes(:event_type).joins(:event_signups).
```

Comment:
This is unlikely to happen but just in case you could add `require_login` just before this user_event assignment so that if a user tries to load this route without logging in it will redirect them to the login page instead of throwing an error since @user.id would be null.